h2integrate/converters/wind/floris.py

```diff
@@ -0,0 +1,299 @@
+        )
+        # check if turbulence intensity is available in wind resource data
+        if any("turbulence_intensity_" in k for k in wind_resource_data.keys()):
+            for height in bounding_heights:
+                if f"turbulence_intensity_{height}m" in wind_resource_data:
+                    ti = wind_resource_data.get(f"turbulence_intensity_{height}m", default_ti)
+                    break
+        else:
+            ti = wind_resource_data.get("turbulence_intensity", default_ti)
+
+        if not isinstance(ti, (int, float)):
+            if len(ti) == 0:
+                ti = 0.06
+            if not isinstance(ti, (float, int)) and len(ti) == 1:
+                ti = float(ti[0])
+            if not isinstance(ti, (float, int)) and len(ti) != self.n_timesteps:
+                msg = (
+                    f"Turbulence intensity is length {len(ti)} but must be "
+                    f"either be length {self.n_timesteps} or a float"
+                )
+                raise ValueError(msg)
+
+        time_series = TimeSeries(
+            wind_directions=winddir,
+            wind_speeds=windspeed,
+            turbulence_intensities=ti,
+        )
+
+        return time_series
+
+    def compute(self, inputs, outputs, discrete_inputs, discrete_outputs):
+        # NOTE: could update air density based on elevation if elevation is included
+        # in the resource data.
+        # would need to duplicate the ``calculate_air_density`` function from HOPP
+
+        # Check if the results for the current configuration are already cached
+        if self.config.enable_caching:
+            config_dict = self.config.as_dict()
+            config_dict.update({"wind_turbine_size_kw": self.wind_turbine_rating_kW})
+            cache_filename = make_cache_hash_filename(
+                config_dict, inputs, discrete_inputs, cache_dir=self.config.cache_dir
+            )
+
+            if Path(cache_filename).exists():
+                # Load the cached results
+                cache_path = Path(cache_filename)
+                with cache_path.open("rb") as f:
+                    cached_data = dill.load(f)
+                outputs["electricity_out"] = cached_data["electricity_out"]
+                outputs["total_capacity"] = cached_data["total_capacity"]
+                outputs["total_electricity_produced"] = cached_data["total_electricity_produced"]
+                outputs["capacity_factor"] = cached_data["capacity_factor"]
+                return
+
+        # If caching is not enabled or a cache file does not exist, run FLORIS
+        n_turbs = int(np.round(inputs["num_turbines"][0]))
+
+        # Copy main config files
+        floris_config = copy.deepcopy(self.config.floris_wake_config)
+        turbine_design = copy.deepcopy(self.config.floris_turbine_config)
+
+        # update the turbine hub-height in the floris turbine config
+        turbine_design.update({"hub_height": inputs["hub_height"][0]})
```

Comment:
This updates the floris config input with the hub height defined in the input file (outside of the floris turbine definition)?